staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -476,15 +476,32 @@ func DefaultKubernetesUserAgent() string {
 		adjustCommit(version.Get().GitCommit))
 }
 
+// getServiceAccountFilePaths returns the paths to the serviceaccount token files
+// In the case of Windows HostProcess containers this prepends the CONTAINER_SANDBOX_MOUNT_POINT env variable
+// for other operating systems or if the sandbox env variable is not set it returns the standard mount points
+// see https://kubernetes.io/docs/tasks/configure-pod-container/create-hostprocess-pod/#volume-mounts
+func getServiceAccountFilePaths() (string, string) {
+	const (
+		tokenFile  = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+		rootCAFile = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+	)
+
+	if gruntime.GOOS == "windows" {
+		sandbox := os.Getenv("CONTAINER_SANDBOX_MOUNT_POINT")
+		if sandbox != "" {
```

Comment:
should this rely on the KEP-defined behavior and blindly prepend the envvar to the path? if it's empty, it'll no-op, and if it's populated, it is expected to hold a path that could be prepended in a manifest, right?

Reply:
do the forward-slash-delimited paths above work as expected on windows, especially when combined with a backslash-delimited $CONTAINER_SANDBOX_MOUNT_POINT var?

per the example in the KEP, prepending would end up with paths looking like this:

`c:\c\<container id>\/var/run/secrets/kubernetes.io/serviceaccount/token`

Reply:
The win32 file APIs accept both (regardless of how strange it is...). I'd had a comment on sort of the same, https://github.com/kubernetes/kubernetes/pull/104490#discussion_r693276579. `filepath.Join` on a Windows host will actually correct it to the "windows" way of doing things, and adjust the unix like path to backslashes anyways. So 	`filepath.Join("C:\\path\\to\\thing", "/hello/world")` -> C:\\path\\to\\thing\\hello\\world, even if the first path ends with a trailing slash or no trailing slash.

Reply:
@liggitt The logic here is basically: For Windows Server containers we don't want to change anything. The above will just get treated as an absolute path on the C drive which is what everyone's used to. If this env var is present, which is our tip off that this is a HostProcess container as 99% of the time I'd imagine that's the only time this would be present, then we want to join the path with where the rootfs for the container is on the host.

Reply:
if simple concatenation work, I think we should limit ourselves to that, since that is what use of the envvar in args, commands, and workingDir is limited to

Reply:
Sure, that sounds reasonable to me

Reply:
We joined here as in the other spots we're taking user input and simply expanding to what the envvar contains. Here, there is none and we're just taking what the previous logic was and having it point to the new location, seemed better to have a more friendly looking path even if the gnarly looking one will work for the file apis.